tools/update.py

```diff
@@ -318,26 +334,19 @@ def _GetPackageFilenamesAndVersions(self, package_names):
         preferred_operating_system=self.operating_system)
     if not package_urls:
       logging.error('Unable to determine package download URLs.')
-      return None, None
-
-    if not os.path.exists(self._download_directory):
-      os.mkdir(self._download_directory)
-
-    os.chdir(self._download_directory)
+      return []
 
-    package_filenames = {}
+    available_packages = {}
```

Comment:
Are you using a dict here because you want to ignore collisions for the same name? I'd prefer an explicit check  that condition here, and change this to a list, rather than returning available_packages.values()

If you still want it this way, please add a comment about why this is a dict.